### PR TITLE
fix(input-field): apply the user's locale when formatting numbers

### PR DESCRIPTION
**Pull request by Befkadu1**
The property "locale" specifies the user's locale and/or preferences in which numbers should be formatted. This affect the formatting of integers and decimals

fix Lundalogik/crm-feature#3673

## Review:
- [x] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/input-field/input-field.tsx [hunk 1]
```diff
@@ -217,6 +217,12 @@ export class InputField {
     @Prop({ reflect: true })
     public showLink = false;
 
+    /**
+     * The user's locale that determines how numbers are formatted.
+     */
+    @Prop({ reflect: true })
+    public locale: string = navigator.language;
```
Comment (vicgeralds) [inline src/components/input-field/input-field.tsx:224]: We could use global config here (`import config from '../../global/config';`)

```suggestion
    public locale: string = config.defaultLocale;
```

The icon component is an example that uses global config, which is provided using `limel-config`

Comment (Befkadu1) [reply src/components/input-field/input-field.tsx:224]: Would it be better to initialize `defaultLocale` to `en`? 

```
export class Config {
    public iconPath = '';
    public defaultLocale: string
    public featureSwitches: any = getFeatureSwitches(localStorage);
}

Comment (vicgeralds) [reply src/components/input-field/input-field.tsx:224]: no, `navigator.language` is better I think, as it's the user's preferred language

**Review by vicgeralds — APPROVED** [review]
Looks good

**Pull request merged**